src/components/FlawHistory/FlawHistory.vue

```diff
@@ -12,11 +13,47 @@ const props = defineProps<{
   history: null | undefined | ZodFlawHistoryItemType[];
 }>();
 
+const startDate = ref<null | string | undefined>(null);
+const endDate = ref<null | string | undefined>(null);
+
+const emptyFilters = computed(() => {
+  return !startDate.value && !endDate.value;
+});
+
+const validDateRange = computed(() => {
+  const start = startDate.value ? new Date(startDate.value) : null;
+  const end = endDate.value ? new Date(endDate.value) : null;
+
+  return (
+    start instanceof Date && !Number.isNaN(start.getTime())
+    && end instanceof Date && !Number.isNaN(end.getTime())
+    && end >= start
+  );
+});
+
+const filteredHistoryItems = computed(() => {
+  if (!validDateRange.value) {
+    return props.history;
+  }
+
+  const start = new Date(startDate.value!);
+  const end = new Date(endDate.value!);
+  end.setDate(end.getDate() + 1);
+
+  return props.history?.filter((item) => {
+    if (!item.pgh_created_at) return false;
+
+    const itemDate = new Date(item.pgh_created_at);
+
+    return itemDate.getTime() >= start.getTime() && itemDate.getTime() <= end.getTime();
```

Comment:
```suggestion
    return itemDate.getTime() >= start.getTime() && itemDate.getTime() < end.getTime();
```

Reply:
I think this needs to just be less than rather than less than or equal to so it doesn't capture UTC 00:00 of the next day